lasagne/layers/normalization.py

```diff
@@ -101,3 +115,259 @@ def get_output_for(self, input, **kwargs):
+            # and applied, but the computation will be optimized away):
+            mean += 0 * running_mean
+            var += 0 * running_var
+
+        # prepare dimshuffle pattern inserting broadcastable axes as needed
+        param_axes = iter(range(input.ndim - len(self.axes)))
+        pattern = ['x' if input_axis in self.axes
+                   else next(param_axes)
+                   for input_axis in range(input.ndim)]
+
+        # apply dimshuffle pattern to all parameters
+        beta = 0 if self.beta is None else self.beta.dimshuffle(pattern)
+        gamma = 1 if self.gamma is None else self.gamma.dimshuffle(pattern)
+        mean = mean.dimshuffle(pattern)
+        std = T.sqrt(var + self.epsilon).dimshuffle(pattern)
+
+        # normalize
+        # normalized = (input - mean) * (gamma / std) + beta
+        normalized = T.nnet.batch_normalization(input, gamma=gamma, beta=beta,
+                                                mean=mean, std=std,
+                                                mode=self.mode)
+        return normalized
+
+
+def batch_norm(layer, **kwargs):
+    """
+    Apply batch normalization to an existing layer. This is a convenience
+    function modifying an existing layer to include batch normalization: It
+    will steal the layer's nonlinearity if there is one (effectively
+    introducing the normalization right before the nonlinearity), remove
+    the layer's bias if there is one (because it would be redundant), and add
+    a :class:`BatchNormLayer` and :class:`NonlinearityLayer` on top.
+
+    Parameters
+    ----------
+    layer : A :class:`Layer` instance
+        The layer to apply the normalization to; note that it will be
+        irreversibly modified as specified above
+    **kwargs
+        Any additional keyword arguments are passed on to the
+        :class:`BatchNormLayer` constructor. Especially note the `mode`
+        argument, which controls a memory usage to performance tradeoff.
+
+    Returns
+    -------
+    BatchNormLayer or NonlinearityLayer instance
+        A batch normalization layer stacked on the given modified `layer`, or
+        a nonlinearity layer stacked on top of both if `layer` was nonlinear.
+
+    Examples
+    --------
+    Just wrap any layer into a :func:`batch_norm` call on creating it:
+
+    >>> from lasagne.layers import InputLayer, DenseLayer, batch_norm
+    >>> from lasagne.nonlinearities import tanh
+    >>> l1 = InputLayer((64, 768))
+    >>> l2 = batch_norm(DenseLayer(l1, num_units=500, nonlinearity=tanh))
+
+    This introduces batch normalization right before its nonlinearity:
+
+    >>> from lasagne.layers import get_all_layers
+    >>> [l.__class__.__name__ for l in get_all_layers(l2)]
+    ['InputLayer', 'DenseLayer', 'BatchNormLayer', 'NonlinearityLayer']
+    """
+    nonlinearity = getattr(layer, 'nonlinearity', None)
+    if nonlinearity is not None:
+        layer.nonlinearity = nonlinearities.identity
+    if hasattr(layer, 'b'):
```

Comment:
If `layer` already has `b` set to None, the line below will throw an exception since `layer.params` will not include the bias. Perhaps the condition should check `layer.b == None` too?